### PR TITLE
Migrate from springfox to springdoc.

### PR DESCRIPTION
**Pull request by mshima**
Depends on https://github.com/jhipster/jhipster-bom/pull/556.

Closes https://github.com/jhipster/generator-jhipster/issues/17175

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/client/templates/common/src/main/webapp/swagger-ui/index.html.ejs [hunk 1]
```diff
@@ -64,9 +64,15 @@
...
-            urls.push({ name: resource.name, url: resource.location });
-          });
+          if (response.data && response.data.forEach) {
+            response.data.forEach(function (resource) {
+              urls.push({ name: resource.name, url: resource.location });
+            });
+          }
+        }, () => {}).then(() => {
+          if (urls.length === 0) {
+        	urls.push({ name: '<%= baseName %>', url: '/v3/api-docs' });
```
Comment (mraible) [inline generators/client/templates/common/src/main/webapp/swagger-ui/index.html.ejs:74]: Indent needed here.

**Review by mraible — APPROVED** [review]

**Comment by pascalgrimaud** [comment]
We need to do some manual test to check if it's ok with microservices+jwt and microservices+oauth2
As this part was a little bit tricky: reactive gateway and mvc microservice :-)

**Comment by mshima** [comment]
> We need to do some manual test to check if it's ok with microservices+jwt and microservices+oauth2
> As this part was a little bit tricky: reactive gateway and mvc microservice :-)

Plan to implement it at client side if there is no problem!?

**Comment by pascalgrimaud** [comment]
I don't know exactly how it's done today, I think there are some part done in backend (routing) and other part done in frontend (for security jwt/oauth2).
Never mind the solution, it should be simple if possible :)

And if we could give the possibility to switch to something like https://github.com/Redocly/redoc it would be perfect.

**Comment by mshima** [comment]
@atomfrede any clue why gradle is behaving different and failing?
Seems to me that it's pulling spring-boot 2.6.0 dependencies.

**Comment by mshima** [comment]
> @atomfrede any clue why gradle is behaving different and failing?
> Seems to me that it's pulling spring-boot 2.6.0 dependencies.

Found the problem.

**Comment by mraible** [comment]
Yay! All tests are passing. 😀

**Comment by Akuka** [comment]
Hi
It looks like something wrong with Criteria after this change, Are you aware?

This is how it looks like:
image

**Comment by mshima** [comment]
@Akuka you should create a new issue, what was expected?
I opened side by side and looked the same.
springdoc broke/fixed a few times after the initial version.

**Comment by Akuka** [comment]
> @Akuka you should create a new issue, what was expected? I opened side by side and looked the same. springdoc broke/fixed a few times after the initial version.

Its never worked
I opened an issue https://github.com/jhipster/generator-jhipster/issues/17776

**Pull request merged**